homeassistant/components/device_tracker/automatic.py

```diff
@@ -0,0 +1,170 @@
+"""
+Support for the Automatic platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.automatic/
+"""
+from datetime import timedelta
+import logging
+import re
+import threading
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle, datetime as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+
+CONF_CLIENT_ID = 'client_id'
+CONF_SECRET = 'secret'
+CONF_DEVICES = 'devices'
+
+SCOPE = 'scope:location scope:vehicle:profile scope:user:profile scope:trip'
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_EXPIRES_IN = 'expires_in'
+ATTR_RESULTS = 'results'
+ATTR_VEHICLE = 'vehicle'
+ATTR_ENDED_AT = 'ended_at'
+ATTR_END_LOCATION = 'end_location'
+
+URL_AUTHORIZE = 'https://accounts.automatic.com/oauth/access_token/'
+URL_VEHICLES = 'https://api.automatic.com/vehicle/'
+URL_TRIPS = 'https://api.automatic.com/trip/'
+
+_VEHICLE_ID_REGEX = re.compile(
+    (URL_VEHICLES + '(.*)?[/]$').replace('/', r'\/'))
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): cv.string,
+    vol.Required(CONF_SECRET): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_DEVICES): vol.All(cv.ensure_list, [cv.string])
+})
+
+
+def setup_scanner(hass, config: dict, see):
+    """Validate the configuration and return an Automatic scanner."""
+    try:
+        AutomaticDeviceScanner(config, see)
+    except IOError:
+        return False
```

Comment:
Maybe add `except IOError as err: _LOGGER.error(str(err))` here to print your exception messages you throw below, else it will just fail silently